### PR TITLE
chore(spanner): fix issues with licensing

### PR DESCRIPTION
**Pull request by bajajneha27**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-spanner/LICENSE [hunk 1]
```diff
@@ -192,7 +192,7 @@
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at
 
-       https://www.apache.org/licenses/LICENSE-2.0
+       http://www.apache.org/licenses/LICENSE-2.0
```
Comment (NivedhaSenthil) [inline google-cloud-spanner/LICENSE:195]: why https to http here ?

File: google-cloud-spanner/LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
                             Apache License
                            Version 2.0, January 2004
-                        https://www.apache.org/licenses/
```
Comment (NivedhaSenthil) [inline google-cloud-spanner/LICENSE:3]: why https to http here ?

**Review by NivedhaSenthil — APPROVED** [review]

File: acceptance/data/index.yaml [hunk 3]
```diff
@@ -1,3 +1,17 @@
+# Copyright 2014 Google LLC
```
Comment (SandeepTuniki) [inline acceptance/data/index.yaml:1]: Just to clarify, does the copyright always points to the year of the file's original creation? Even when it is added retroactively?

File: google-cloud-spanner/LICENSE (hunk 1 above)
Comment (bajajneha27) [reply google-cloud-spanner/LICENSE:195]: Responded offline

File: acceptance/data/index.yaml (hunk 3 above)
Comment (bajajneha27) [reply acceptance/data/index.yaml:1]: Ideally, you should add this when you create the file.

**Review by dazuma — APPROVED** [review]
Thanks!

**Pull request merged**